### PR TITLE
Add MicroMod STM32WB5MMG. Move MicroMod STM32F405 under SparkFun Boards file.

### PR DESCRIPTION
**Pull request by SFE-Brudnerd**
**Summary**

Adds the MicroMod Processor Board - STM32WB5MMG board files.

Consolidates the SparkFun STM32 products under one menu.

This PR fixes/implements the following **features**

* Adds the MicroMod Processor Board - STM32WB5MMG

Explain the **motivation** for making this change. What existing problem does the pull request solve?

The motivation of the change is to add Arduino support for an upcoming board release.

**Validation**

* Ensure CI build is passed. ✅ 

**Code formatting**

* Ensure AStyle check is passed thanks CI ✅

### CODE REVIEW

**Review by fpistm — CHANGES_REQUESTED** [review]
Hi @SFE-Brudnerd 
Thanks for this PR.
You can find here my first remarks about this PR. I will deeply check later.

File: README.md [hunk 1]
```diff
@@ -49,39 +49,41 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
...
+- [Blues Wireless boards](#blues-wireless-boards)
+- [Elecgator boards](#elecgator-boards)
+- [Electronic Speed Controller boards](#electronic-speed-controller-boards)
+- [Garatronic/McHobby boards](#garatronicmchobby-boards)
+- [Generic flight controllers](#generic-flight-controllers)
+- [LoRa boards](#lora-boards)
+- [Midatronics boards](#midatronics-boards)
+- [SparkFun boards](#sparkfun-boards)
+- [Next release](#next-release)
+- [Troubleshooting](#troubleshooting)
```
[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply README.md:86]: Agreed, I was having some fun with VSCode's auto ToC feature yesterday, I must have missed this bit. Good catch.

[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply README.md]: Still to do with the formatter auto-changing on save. Thanks!

File: README.md [hunk 2]
```diff
@@ -775,6 +776,13 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :----: | :-------: | ---- | :-----: | :---- |
 | :green_heart:  | STM32WB55CE | [SharkyMKR](https://midatronics.com/shop/development-boards/mkr-sharky-i/) | *1.7.0* |  |
 
+### [SparkFun](https://www.sparkfun.com/) boards
+
+| Status | Device(s) | Name | Release | Notes |
+| :----: | :-------: | ---- | :-----: | :---- |
+| :yellow_heart: | STM32F405RG | [SparkFun MicroMod Processor Board - STM32F405](https://www.sparkfun.com/products/21326) | **2.6.0** |  |
+| :yellow_heart: | STM32WB5MMG | [SparkFun MicroMod Processor Board - STM32WB5MMG](https://www.sparkfun.com/products/21438) | **2.6.0** |  |
```
[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply README.md:782]: Yes, the product is yet unreleased so it isn't live. However, it is the valid link.

File: README.md [hunk 3]
```diff
@@ -324,8 +326,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
-| :green_heart:  | STM32F405RG | [Adafruit Feather STM32F405 Express](https://www.adafruit.com/product/4382) | *1.8.0* |  |
-| :yellow_heart:  | STM32F405RG | [SparkFun MicroMod STM32](https://www.sparkfun.com/products/21326) | **2.6.0** |  |
+| :green_heart:  | STM32F405RG | [Adafruit Feather STM32F405 Express](https://www.adafruit.com/product/4382) | *1.8.0* |
```
[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply README.md:327]: In the boards file I had moved the SparkFun MMPB F405 into the SparkFun section I created, so I figured it was good for these lists to reflect the boards file.

[the remark this answers is not in the document]
Comment (fpistm) [reply README.md:327]: No you removed a pipe '|'. So the line should bd restored.

[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply README.md:327]: Oh, I missed that. Apologies! I'll make that change.

**Comment by SFE-Brudnerd** [comment]
Hey @fpistm, when I was running the arduino-cli workflow, I received [this error](https://github.com/SFE-Brudnerd/Arduino_Core_STM32/actions/runs/5205220874/jobs/9390432598#step:5:396), from what I'm reading this is an issue outside of the changes I've made. Is this expected?

**Comment by fpistm** [comment]
No. It seems your changes brings issue with {} issue. Will check that.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (fpistm) [reply README.md:782]: When it will be valid ?

**Comment by fpistm** [comment]
Let's CI run to have a first result. :wink:

File: cmake/boards_db.cmake [hunk 4]
```diff
@@ -104645,6 +104600,91 @@ target_compile_options(RUMBA32_xusb_HSFS INTERFACE
...
+  ${CMAKE_CURRENT_LIST_DIR}/../system/Drivers/STM32WBxx_HAL_Driver/Inc
+  ${CMAKE_CURRENT_LIST_DIR}/../system/Drivers/STM32WBxx_HAL_Driver/Src
+  ${CMAKE_CURRENT_LIST_DIR}/../system/Drivers/CMSIS/Device/ST/STM32WBxx/Include/
+  ${CMAKE_CURRENT_LIST_DIR}/../system/Drivers/CMSIS/Device/ST/STM32WBxx/Source/Templates/gcc/
+  ${SFE_MMPB_STM32WB5MMG_VARIANT_PATH}
+)
+
+target_link_options(SFE_MMPB_STM32WB5MMG INTERFACE
+  "LINKER:--default-script=${SFE_MMPB_STM32WB5MMG_VARIANT_PATH}/ldscript.ld"
+  "LINKER:--defsym=LD_FLASH_OFFSET=0"
```
Comment (fpistm) [inline cmake/boards_db.cmake:104637]:
```suggestion
  "LINKER:--defsym=LD_FLASH_OFFSET=0x0"
```

**Review by fpistm — CHANGES_REQUESTED** [review]
Build issue comes from the flash_offset not defined in boards.txt

**Review by fpistm — CHANGES_REQUESTED** [review]

File: cmake/boards_db.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply cmake/boards_db.cmake:104637]: image
It won't let me make the change... It's been doing this for hours now.

[the remark this answers is not in the document]
Comment (fpistm) [reply cmake/boards_db.cmake:104637]: No worry. It will be updated next time I will update the database.

**Comment by fpistm** [comment]
You didn't try to build before? So you didn't make any tests with the board?

**Comment by SFE-Brudnerd** [comment]
I did a fair bit of testing before I pulled the upstream in, I realized that I didn't test afterwards.

File: cmake/boards_db.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (SFE-Brudnerd) [reply cmake/boards_db.cmake:104637]: Commit #[4df0037](https://github.com/stm32duino/Arduino_Core_STM32/pull/2037/commits/4df00371e024e110de011ee5b5a1ec49259cbf46) solves this, not sure why it still shows as unresolved.

File: README.md [hunk 5]
```diff
@@ -775,6 +774,13 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :----: | :-------: | ---- | :-----: | :---- |
 | :green_heart:  | STM32WB55CE | [SharkyMKR](https://midatronics.com/shop/development-boards/mkr-sharky-i/) | *1.7.0* |  |
 
+### [SparkFun](https://www.sparkfun.com/) boards
+
+| Status | Device(s) | Name | Release | Notes |
+| :----: | :-------: | ---- | :-----: | :---- |
+| :yellow_heart: | STM32F405RG | [SparkFun MicroMod Processor Board - STM32F405](https://www.sparkfun.com/products/21326) | **2.6.0** |  |
+| :yellow_heart: | STM32WB5MMG | [SparkFun MicroMod Processor Board - STM32WB5MMG](https://www.sparkfun.com/products/21438) | **2.6.0** |  |
```
Comment (fpistm) [inline README.md:782]:
```suggestion
| :yellow_heart: | STM32WB5MMG | SparkFun MicroMod Processor Board - STM32WB5MMG | **2.6.0** |  |
```

**Review by fpistm — CHANGES_REQUESTED** [review]
While the link is not valid, it could not be listed here for security purpose.

File: README.md (hunk 5 above)
Comment (SFE-Brudnerd) [reply README.md:782]: I'll add the link back in when it's live. Thanks!

File: variants/STM32WBxx/WB5MMGH/variant_SFE_MMPB_STM32WB5MMG.h [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+#ifndef TIMER_TONE
+  #define TIMER_TONE            TIM16
+#endif
+#ifndef TIMER_SERVO
+  #define TIMER_SERVO           TIM17
+#endif
+
+// UART Definitions
+#ifndef SERIAL_UART_INSTANCE
+  #define SERIAL_UART_INSTANCE  101
```
Comment (fpistm) [inline variants/STM32WBxx/WB5MMGH/variant_SFE_MMPB_STM32WB5MMG.h:132]:
```suggestion
  #define SERIAL_UART_INSTANCE  1
```

**Review by fpistm — CHANGES_REQUESTED** [review]

**Review by fpistm — APPROVED** [review]

**Comment by fpistm** [comment]
@SFE-Brudnerd 
FYI, I've made a little fix on the `SERIAL_UART_INSTANCE  `which is 1 for `USART1`.  101 is for `LPUART1`.

**Comment by SFE-Brudnerd** [comment]
Works for me, thanks!

**Pull request merged**